### PR TITLE
JDK-8301224: Replace NULL with nullptr in share/gc/shared/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/gc/shared/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301224](https://bugs.openjdk.org/browse/JDK-8301224): Replace NULL with nullptr in share/gc/shared/

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [5369babe](https://git.openjdk.org/jdk/pull/12249/files/5369babef911e4328b8791ae898647559a44960f)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12249/head:pull/12249` \
`$ git checkout pull/12249`

Update a local copy of the PR: \
`$ git checkout pull/12249` \
`$ git pull https://git.openjdk.org/jdk pull/12249/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12249`

View PR using the GUI difftool: \
`$ git pr show -t 12249`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12249.diff">https://git.openjdk.org/jdk/pull/12249.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/c2/barrierSetC2.cpp
Comment (jdksjolen) [inline src/hotspot/share/gc/shared/c2/barrierSetC2.cpp:366]: NULL_PTR

**Review by jdksjolen — COMMENTED** [review]
Found only one issue.

**Review by stefank — APPROVED** [review]

**Comment by jdksjolen** [comment]
Passes tier1.

File: src/hotspot/share/gc/shared/c1/barrierSetC1.cpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/c1/barrierSetC1.cpp:227]: This "null" in a comment really should be "nullptr".

File: src/hotspot/share/gc/shared/generationCounters.cpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/generationCounters.cpp:70]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:102]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:107]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:123]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:128]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:134]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:158]: nullptr here

File: src/hotspot/share/gc/shared/oopStorageParState.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorageParState.hpp:127]: nullptr here, or "*p is null".

**Review by kimbarrett — CHANGES_REQUESTED** [review]
There are a number of changes to something like `*p == null` that I think should either be `*p == nullptr` or `*p is null`.

**Comment by jdksjolen** [comment]
Hi Kim, thanks, I've pushed your suggested changes.

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:107]: Inconsistent mix of "null" and "nullptr".

**Review by kimbarrett — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/shared/oopStorage.hpp
Comment (jdksjolen) [reply src/hotspot/share/gc/shared/oopStorage.hpp]: Thanks!

**Comment by jdksjolen** [comment]
This is still passing tier1, just waiting for approval.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by jdksjolen** [comment]
Passes tier1.

/integrate

**Pull request closed without merging**
